app/_src/gateway/plugin-development/pdk/kong.vault.md

```diff
@@ -12,6 +12,297 @@ source_url: https://github.com/Kong/kong/tree/master/kong/pdk
+## get_subfield(value, key)
+
+This function extracts a subfield from a JSON object.
+ It first decodes the JSON string into a Lua table, then checks for the presence and type of a specific key.
+
+
+**Parameters**
+
+* **value** :  The JSON string to be parsed and decoded.
+* **key** :  The specific subfield to be searched for within the JSON object.
+
+**Returns**
+
+*   On success, returns the value associated with the specified key in the JSON object.
+ If the key does not exist or its value is not a string, returns nil along with an error message.
+ If the input value cannot be parsed as a JSON object, also returns nil along with an error message.
+
+
+
+
+## adjust_ttl(ttl, vault_config)
+
+This function adjusts the 'time-to-live' (TTL) according to the configuration provided in 'vault_config'.
+ If the TTL is not a number or if it falls outside of the configured minimum or maximum TTL, it will be adjusted accordingly.
+
+
+**Parameters**
+
+* **ttl** :  The initial time-to-live value.
+* **vault_config** :  The configuration table for the vault, which may contain 'ttl', 'min_ttl', and 'max_ttl' fields.
+
+**Returns**
+
+*   Returns the adjusted TTL. If the initial TTL is not a number, it returns the 'ttl' field from the 'vault_config' table or 0 if it doesn't exist.
+ If the initial TTL is greater than 'max_ttl' from 'vault_config', it returns 'max_ttl'.
+ If the initial TTL is less than 'min_ttl' from 'vault_config', it returns 'min_ttl'.
+ Otherwise, it returns the original TTL.
+
+
+
+
+## get_vault(prefix)
+
+This function retrieves a vault by its prefix.  It either fetches the vault from a cache or directly accesses it.
+ The vault is expected to be found in a database (db) or cache. If not found, an error message is returned.
+
+
+**Parameters**
+
+* **prefix** :  The unique identifier of the vault to be retrieved.
+
+**Returns**
+
+*  Returns the vault if it's found. If the vault is not found, it returns nil along with an error message.
+
+
+
+
+## get_vault_strategy_and_schema_during_init(name)
+
+Fetches the strategy and schema for a given vault during initialization.
+
+ This function checks if the vault exists in `VAULT_NAMES`, fetches the associated strategy and schema from
+ the `STRATEGIES` and `SCHEMAS` tables, respectively. If the strategy or schema isn't found in the tables, it
+ attempts to fetch them from the application's database or by requiring them from a module.
+
+ The fetched strategy and schema are then stored back into the `STRATEGIES` and `SCHEMAS` tables for later use.
+ If the `init` method exists in the strategy, it's also executed.
+
+
+**Parameters**
+
+* **name** :  string The name of the vault to fetch the strategy and schema for.
+
+**Returns**
+
+1.   strategy ??? The fetched or required strategy for the given vault.
+
+1.   schema ??? The fetched or required schema for the given vault.
+
+1.   string|nil An error message, if an error occurred while fetching or requiring the strategy or schema.
```

Comment:
The formatting in this section looks strange. Compare it to https://github.com/Kong/docs.konghq.com/pull/5882/files#diff-40cdd2ce6cfefad5924186029bcd097d256de1daf1ff081d16211a3779e299feR140-R145